qc_tests/streaks.py

```diff
@@ -203,29 +532,50 @@ def rsc(station, var_list, config_dict, full=False, plots=False, diagnostics=Fal
     for var in var_list:
 
         obs_var = getattr(station, var)
+        if obs_var.name == "wind_direction":
+            wind_speed = copy.deepcopy(getattr(station, "wind_speed"))
+        else:
+            wind_speed = None
 
         # need to have at least two observations to get a streak
         if len(obs_var.data.compressed()) >= 2:
-
-            # repeating (straight) strings
             if full:
-                get_repeating_string_threshold(obs_var, config_dict, plots=plots, diagnostics=diagnostics)
-
-            repeating_value(obs_var, station.times, config_dict, plots=plots, diagnostics=diagnostics)
-
-            # more short strings than reasonable
-            # excess_repeating_value()
+                # recalculating all thresholds
+                get_repeating_streak_threshold(obs_var, config_dict, wind_speed=wind_speed,
+                                               plots=plots, diagnostics=diagnostics)
+                get_excess_streak_threshold(obs_var, station.years, config_dict, wind_speed=wind_speed,
+                                            plots=plots, diagnostics=diagnostics)
+
+                # This is slow - as spinning through each day and then selecting
+                repeating_day(obs_var, station, config_dict, determine_threshold=True,
+                              plots=plots, diagnostics=diagnostics)
+
```

Comment:
Double check indentation here. I think its right but its hard to understand the logic a bit.